ocs_ci/ocs/resources/storage_client.py

```diff
@@ -561,3 +553,20 @@ def get_storageclient_name(self, namespace, timeout=300, sleep=10):
                     log.info(f"storageclient data, {storageclient}")
                     return storageclient.get("metadata", {}).get("name")
         return None
+
+    def verify_version_of_odf_client_operator(self, only_major_minor=True):
+        """
+        This method verifies the odf client operator version is as per provider odf version
+
+        """
+        ocs_csv = get_ocs_csv()
+        client_csv_version = ocs_csv.data["spec"]["version"]
+        ocs_version = version.get_ocs_version_from_csv(
+            only_major_minor=only_major_minor
+        )
+        log.info(
+            f"Check if OCS version: {ocs_version} matches with CSV: {client_csv_version}"
+        )
+        assert (
+            f"{ocs_version}" in client_csv_version
+        ), f"OCS version: {ocs_version} mismatch with CSV version {client_csv_version}"
```

Comment:
@jilju  @amr1ta once a provider is upgraded, how much time does it take for the client to auto-upgrade?
Also, IIRC this check is for minor version . What is the observation for z-stream upgrades of providers?